File: AWS_GUI_Dev/BAWGUI.Settings/ViewModels/EnumerationManager.vb
Imports System.Collections.ObjectModel
Imports System.ComponentModel
Imports System.Globalization
Imports System.Reflection
Imports System.Linq
Imports System.Windows.Data
Imports System.Windows
Imports System.Drawing
Namespace ViewModels
    Public Class EnumerationManager
        Public Shared Function GetValues(enumeration As Type) As Array
            Dim wArray As Array = [Enum].GetValues(enumeration)
            Dim wFinalArray As New ArrayList()
            For Each wValue As [Enum] In wArray
                Dim fi As FieldInfo = enumeration.GetField(wValue.ToString())
                If fi IsNot Nothing Then
                    Dim wBrowsableAttributes As BrowsableAttribute() = TryCast(fi.GetCustomAttributes(GetType(BrowsableAttribute), True), BrowsableAttribute())
                    If wBrowsableAttributes.Length > 0 Then
                        '  If the Browsable attribute is false
                        If wBrowsableAttributes(0).Browsable = False Then
                            ' Do not add the enumeration to the list.
                            Continue For
                        End If
                    End If

                    Dim wDescriptions As DescriptionAttribute() = TryCast(fi.GetCustomAttributes(GetType(DescriptionAttribute), True), DescriptionAttribute())
                    If wDescriptions.Length > 0 Then
                        wFinalArray.Add(wDescriptions(0).Description)
                    Else
                        wFinalArray.Add(wValue.ToString())
                    End If
                End If
            Next
            Return wFinalArray.ToArray()
        End Function
    End Class
End Namespace

File: AWS_GUI_Dev/BAWGUI.Utilities/EnumerationManager.cs
using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.Globalization;
using System.IO;
using System.Linq;
using System.Reflection;
using System.Runtime.CompilerServices;
using System.Security;
using System.Text;
using System.Threading.Tasks;
using Microsoft.VisualBasic;
using System.Collections.ObjectModel;
using System.ComponentModel;
using System.Windows.Data;
using System.Windows;
using System.Drawing;
using System.Collections;

namespace BAWGUI.Utilities
{
    public class EnumerationManager
    {
        public static Array GetValues(Type enumeration)
        {
            Array wArray = Enum.GetValues(enumeration);
            ArrayList wFinalArray = new ArrayList();
            foreach (Enum wValue in wArray)
            {
                FieldInfo fi = enumeration.GetField(wValue.ToString());
                if (fi != null)
                {
                    BrowsableAttribute[] wBrowsableAttributes = fi.GetCustomAttributes(typeof(BrowsableAttribute), true) as BrowsableAttribute[];
                    if (wBrowsableAttributes.Length > 0)
                    {
                        // If the Browsable attribute is false
                        if (wBrowsableAttributes[0].Browsable == false)
                            // Do not add the enumeration to the list.
                            continue;
                    }

                    DescriptionAttribute[] wDescriptions = fi.GetCustomAttributes(typeof(DescriptionAttribute), true) as DescriptionAttribute[];
                    if (wDescriptions.Length > 0)
                        wFinalArray.Add(wDescriptions[0].Description);
                    else
                        wFinalArray.Add(wValue.ToString());
                }
            }
            return wFinalArray.ToArray();
        }
    }
}
